ecosystem/node-checker/src/args.rs

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{metric_evaluator::StateSyncMetricsEvaluatorArgs, runner::BlockingRunnerArgs};
+use clap::Parser;
+use once_cell::sync::Lazy;
+use url::Url;
+
+pub const DEFAULT_METRICS_PORT: u16 = 9101;
+pub const DEFAULT_API_PORT: u16 = 8080;
+pub const DEFAULT_NOISE_PORT: u16 = 6180;
+
+pub static DEFAULT_METRICS_PORT_STR: Lazy<String> =
+    Lazy::new(|| format!("{}", DEFAULT_METRICS_PORT));
+pub static DEFAULT_API_PORT_STR: Lazy<String> = Lazy::new(|| format!("{}", DEFAULT_API_PORT));
+pub static DEFAULT_NOISE_PORT_STR: Lazy<String> = Lazy::new(|| format!("{}", DEFAULT_NOISE_PORT));
+
+#[derive(Clone, Debug, Parser)]
+#[clap(author, version, about, long_about = None)]
+pub struct Args {
+    /// Whether to enable debug logging or not. This is a shortcut for the
+    /// standard env logger configuration via env vars
+    #[clap(short, long)]
+    pub debug: bool,
+
+    /// What address to listen on.
+    #[clap(long, default_value = "0.0.0.0")]
+    pub listen_address: String,
+
+    /// What port to listen on.
+    #[clap(long, default_value = "20121")]
+    pub listen_port: u16,
+
+    /// The URL of the baseline node, e.g. http://fullnode.devnet.aptoslabs.com
+    /// We assume this is just the base and will add ports and paths to this.
+    #[clap(long)]
+    pub baseline_node_url: Url,
+
+    /// The metrics port for the baseline node.
+    #[clap(long, default_value = &DEFAULT_METRICS_PORT_STR)]
+    pub baseline_metrics_port: u16,
+
+    /// The API port for the baseline node.
+    #[clap(long, default_value = &DEFAULT_API_PORT_STR)]
+    pub baseline_api_port: u16,
+
+    /// The port over which validator nodes can talk to the baseline node.
+    #[clap(long, default_value = &DEFAULT_NOISE_PORT_STR)]
+    pub baseline_noise_port: u16,
+
+    /// If this is given, the user will be able to call the check_preconfigured_node
+    /// endpoint, which takes no target, instead using this as the target. If
+    /// allow_test_node_only is set, only the todo endpoint will work,
```

Comment:
- `only the todo endpoint will work`? What does this mean?
- Also `allow_test_node_only`? This should be, `allow_preconfigured_test_node_only`, right?